src/ray/core_worker/experimental_mutable_object_provider.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include "ray/core_worker/common.h"
+#include "ray/core_worker/experimental_mutable_object_manager.h"
+#include "ray/core_worker/store_provider/plasma_store_provider.h"
+#include "ray/raylet_client/raylet_client.h"
+#include "ray/rpc/client_call.h"
+#include "ray/rpc/worker/core_worker_client_pool.h"
+
+namespace ray {
+namespace core {
+namespace experimental {
+
+class MutableObjectProvider {
```

Comment:
Could you add a short doc here to explain what the class is for?

Also, is it only for mutable objects sent over the network? It might be good to include that in the name somehow.

Reply:
Done. Could you elaborate on the change you want to see to the name?

Reply:
I think something like RemoteMutableObjectProvider would be good.

Actually, though, I wonder if this class should be handling both local and remote objects? (see other comments)

Reply:
Done.